src/amuse/community/brutus/interface.py

```diff
@@ -182,12 +318,131 @@ def __init__(self, convert_nbody = None, **options):
             convert_nbody,
             **options
         )
+        self.convert_nbody=convert_nbody
+        self.adjust_prec()
+            
+    def adjust_prec(self):
+        len = self.parameters.word_length
+        if (len > mpmath.mp.prec):
+            mpmath.mp.prec=len
+        
     
     def initialize_code(self):
         result = self.overridden().initialize_code()
         self.parameters.brutus_output_directory = self.output_directory
         return result
     
+    def get_potential_energy_p_si(self):
+        self.adjust_prec()
+        a=self.convert_nbody.to_si(nbody_system.energy).number
+        b=mpmath.mpf(self.get_potential_energy_string())*a
+        return b #* self.convert_nbody.to_si(nbody_system.energy).unit
```

Comment:
I would put this, and the other get_.._p_si in brutusInterface instead; then they should not need the nbody conversion (can be named get_.._mpreal instead)

Reply:
also: why is the adjust_precision necessary?

Reply:
Hello,
you have recognized "import mpmath"? The actual Brutusinterface is not able to handle mpmath. If you want to integrate a complete mpmath handling to your interface I'd welcome it. But for the moment I can't do this. Therefore I put the conversion to the driver to not impact all parts of the system. The adjustment is nessesary to get same precision on both sides of the interface ( Python-mpmath; C++-mpreal)

Reply:
yea, I saw it...I think its ok for now..I am not a 100% sure how mpmath manages the precision in its context..it could also be put in the setter of the precision of Brutus and/or commit_parameters..but don't worry about that..

Reply:
mpmath allows you to increase and decrease see  http://mpmath.org/doc/current/basics.html?highlight=prec#temporarily-changing-the-precision. But I do just increase. The other direction is not of interest. I think this more easy than handling different setters.